tests/__init__.py

```diff
@@ -200,7 +201,7 @@ def test_something(self, g):
             time_hi = min(
                 b.data["demes"][j]["start_time"], b.data["demes"][k]["start_time"]
             )
-            if time_hi - time_lo < 1e308:
```

Comment:
Haha, quite a big difference here! Perhaps a sign should be using a constant here, e.g. from  numpy.finfo

Reply:
Comparing to a constant is pretty weak, we should be using math.isclose or something anyway.

Reply:
The problem here is that if we generate values for `time_hi` and `time_lo`, then later draw more numbers in the open interval `(time_hi, time_lo)`. So if `time_hi` and `time_lo` are too close, hypothesis gives an error that there aren't any floating point numbers within that range. Any idea which particular floating point constant is appropriate here? I *think* the spacing between floating point numbers differs across the range, so I don't know if there is one number that applies. Some small delta should be ok though (and maybe that's bigger than `1e-308`)?

Reply:
Hmm, what about  if ``time_hi == np.nextafter(time_lo)``? That tells you there's no gap between them. https://numpy.org/doc/stable/reference/generated/numpy.nextafter.html#numpy.nextafter

Reply:
Ok, I'll rip that commit out and do it properly as a separate PR. It fails on my workstation, but I haven't seen it fail in CI yet.